### DISCUSSION TITLE
Audit trail

### DISCUSSION THREAD

**Discussion by sussexrick** [umbraco/Umbraco.Forms.Issues#1500] [Ideas]
Forms would support an audit trail in the form designer, similar to the one for content. This would be useful because when you have multiple people designing forms, you often need to know who's been working on it.

**Comment by nul800sebastiaan** [comment]
Hi Rick, we don't have immediate plans to do this but I've put this on the "idea" list for the future when we work on auditing features.

For now you could build this yourself by hooking into `FormStorage.Saving` / `.Saved` events for example.

**Comment by millnut** [comment]
👍 for me, we get alot of requests for this for forms due to content already having this

**Comment by bjarnef** [comment]
@rickbutterfield recently we have use forms for payment and checkout form, where it would be great to log in audits, which changes happens over time. Perhaps event rollback in future.

Another option is to use `transferFormsAsContent` to have changes in git similar to dictionary items, but generally speaking I don't like this approach to entities where content editors have access to:
https://docs.umbraco.com/umbraco-deploy/deployment-workflow/content-transfer#umbraco-forms
